### DISCUSSION TITLE
How can I use "s" for substitution rather than search?

### DISCUSSION THREAD

**Discussion by caffeinism** [LazyVim/LazyVim#613] [Q&A] [answered]
I'd like to use "s" in the substitution function like normal vim, but I can't find a way..

**Comment by folke** [comment, marked as answer]
Disable leap.nvim

**Reply by caffeinism** [reply]
thanks a lot

**Reply by PabloDinella** [reply]
Do people that uses leap.nvim map the default "s" to another key? Or how do they usually replace a character or an entire line ("S")? Just use "vc" and "Vc"? Just wondering

**Reply by folke** [reply]
replace character: `r`

replace whole line `cc`

**Reply by proehlen** [reply]
> Disable leap.nvim

For people coming to this in 2025, I also had to disable flash.  My `~/.config/nvim/lua/plugins/disabled.lua` file:

```
return {
  { "ggandor/leap.nvim", enabled = false },
  { "ggandRE/lf.nvim", enabled = false },
  { "folke/flash.nvim", enabled = false },
}
```

I'm not sure if it was necessary to disable `lf.nvim` but in my searching I found something that suggested it was also a problem.

I appreciate the work y'all have done on LazyVim but some of us have decades of muscle memory using "s" for substitute on vi, vim, vscode with vim bindings and neovim. Some of also have to switch back and forth between those different editors regularly (although vi is becoming less and less common to encounter the wild).

**Comment by jeffsaremi** [comment]
Who came up with the idea of replacing `s` with something else? after having had that for 50 years

**Reply by caffeinism** [reply]
fr 😂

**Reply by bulletmark** [reply]
> Who came up with the idea of replacing `s` with something else? after having had that for 50 years

I've been using vi/vim nearly every day for the last 38 years so it is extremely difficult to stop using `s` and `S`. Still trying to train my brain to use `cl` and `cc` instead. I'm wanting to force myself to leap/sneak around the screen because I can see it is faster to navigate.
